### PR TITLE
remove DocMapper trait

### PR DESCRIPTION
**Pull request by PSeitz**
remove the DocMapper trait and rename `DefaultDocMapper` to `DocMapper`

### CODE REVIEW

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs [hunk 1]
```diff
@@ -61,7 +61,7 @@ const FIELD_PRESENCE_FIELD: Field = Field::from_field_id(0u32);
 /// The mains rules are defined by the field mappings.
 #[derive(Clone, Serialize, Deserialize)]
 #[serde(into = "DefaultDocMapperBuilder", try_from = "DefaultDocMapperBuilder")]
-pub struct DefaultDocMapper {
```
Comment (fulmicoton) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs:64]: we probably should rename the file too?

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs [hunk 2]
```diff
@@ -133,8 +133,8 @@ fn validate_timestamp_field(
     Ok(())
 }
 
-impl From<DefaultDocMapper> for DefaultDocMapperBuilder {
-    fn from(default_doc_mapper: DefaultDocMapper) -> Self {
+impl From<DocMapper> for DefaultDocMapperBuilder {
```
Comment (fulmicoton) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs:136]: And we need to rename `DefaultDocMapperBuilder` too?

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/default_mapper_builder.rs [hunk 3]
```diff
@@ -48,7 +53,7 @@ impl Default for DefaultDocMapperBuilder {
 impl DefaultDocMapperBuilder {
```
Comment (fulmicoton) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/default_mapper_builder.rs:53]: rename struct and file

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/default_mapper_builder.rs [hunk 4]
```diff
@@ -36,6 +36,11 @@ pub struct DefaultDocMapperBuilder {
     /// Default search field names.
     #[serde(default)]
     pub default_search_fields: Vec<String>,
+
+    /// Allow the "type" field separately.
+    /// This is a residue from when the DocMapper was a trait.
+    #[serde(rename = "type", default)]
+    pub legacy_type_tag: Option<String>,
```
Comment (fulmicoton) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/default_mapper_builder.rs:43]: https://docs.rs/serde/latest/serde/de/struct.IgnoredAny.html

Comment (fulmicoton) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/default_mapper_builder.rs:43]:
```suggestion
    #[serde(rename = "type")]
    pub legacy_type_tag: IgnoredAny,
```

**Review by fulmicoton — CHANGES_REQUESTED** [review]
missing some renames

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs (hunk 1 above)
Comment (PSeitz) [reply quickwit/quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs:64]: renamed the file to `doc_mapper_impl`

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs (hunk 2 above)
Comment (PSeitz) [reply quickwit/quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs]: renamed to `DocMapperBuilder`

**Review by fulmicoton — APPROVED** [review]

**Pull request merged**